### ISSUE TITLE
Support for FCStd format

### ISSUE THREAD

**Issue by khimaros** [kovacsv/Online3DViewer#307]
**Describe the solution you'd like**

FCStd is the closest we have to an open standard for CAD project files. the format is a ZIP file with the geometry stored as BRep. the file format is documented here: https://wiki.freecadweb.org/File_Format_FCStd

**Comment by kovacsv** [comment]
Depends on: https://github.com/kovacsv/Online3DViewer/issues/310.

**Comment by khimaros** [comment]
@kovacsv really impressive work and you finished so quickly.

i'm looking forward to seeing this deployed at https://3dviewer.net

i think this starts to give some competition to the online autodesk 360 viewer https://a360.co/2PEZBXG

there hasn't been a convergence on one "assembly" standard for FreeCAD.

@realthunder is leading my personal favorite which is Assembly3.

**Comment by kovacsv** [comment]
Thank, @khimaros! I tested it on a lot of files, but there may be some errors. Here is a teaser image. 🙂

![fcstd_example](https://user-images.githubusercontent.com/2034315/194769691-24a1a40d-45a1-4a96-9ab2-5988b3c72734.png)

**Comment by khimaros** [comment]
i built and ran from the dev branch to test this out. it works great for all of the individual files i loaded. even ones with extremely complex geometry are very performant when loaded. nice work!!

however, none of the assemblies i tried worked. FreeCAD has not settled on a single assembly system, but an underlying primitive [App Link](https://wiki.freecadweb.org/App_Link) has emerged as a standard and is used by most of the assembly workbenches. it allows the same geometry to be used multiple times, across multiple files if desired.

here's the [simplest assembly](https://github.com/kovacsv/Online3DViewer/files/9743184/simple-assembly.zip) that i could come up with to demonstrate the failu more. this is made with the Assembly3 workbench.

there are two test cases:
 - `simple-assembly-unified.FCStd` is a standalone assembly that uses App Link within a single file.
 - `simple-assembly.FCStd` and `simple-cube.FCStd` are used to demonstrate cross-file links.

i can open a separate issue for this if you like because i'd say this is a distinct feature from simple FCStd loading and would understand if it doesn't quite hit the priority bar for you :)

**Comment by khimaros** [comment]
i tried to upload an image of how the assemblies should look, but i'm getting CORS origin errors.

i'll try again tomorrow if that would be helpful but basically it should be two equal sized cubes side by side (attached).

**Comment by khimaros** [comment]
App::Link support tracked as https://github.com/kovacsv/Online3DViewer/issues/312
